pkg/noun/allocate.c

```diff
@@ -2734,11 +2740,19 @@ u3a_loom_sane()
       u3a_fbox *pre_u = u3to(u3a_fbox, this_u->pre_p)
         ,      *nex_u = u3to(u3a_fbox, this_u->nex_p);
 
-      if (nex_p && nex_u->pre_p != this_p) c3_assert(!"loom: wack");
-      if (pre_p && pre_u->nex_p != this_p) c3_assert(!"loom: wack");
+      if (nex_p && nex_u->pre_p != this_p) {
+        fprintf(stderr, "loom: corrupt\r\n");
+        exit(ENOTRECOVERABLE);
+      }
+      if (pre_p && pre_u->nex_p != this_p) {
+        fprintf(stderr, "loom: corrupt\r\n");
+        exit(ENOTRECOVERABLE);
+      }
       if (!pre_p                /* this must be the head of a freelist */
-          && u3R->all.fre_p[_box_slot(this_u->box_u.siz_w)] != this_p)
-        c3_assert(!"loom: wack");
+          && u3R->all.fre_p[_box_slot(this_u->box_u.siz_w)] != this_p) {
```

Comment:
please keep "wack" and don't replace with "corrupt". This makes it easy to differentiate typical corruption errors from those found when the freelist is traced